From 23f617be6412996ddce4fe24621eb32877cbee3f Mon Sep 17 00:00:00 2001
From: Arik Nemtsov <arik@wizery.com>
Date: Tue, 16 Aug 2011 14:15:50 +0300
Subject: [PATCH 2/9] hostap: allow action frames with unknown BSSID in GO mode

P2P invitation responses are transmitted with the BSSID set to the peer
address. Pass these action frames up to prevent the GO from getting stuck
in an infinite invite loop.

Signed-off-by: Arik Nemtsov <arik@wizery.com>
---
 src/ap/ieee802_11.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/ap/ieee802_11.c b/src/ap/ieee802_11.c
index 0ed29b1..23ccee4 100644
--- a/src/ap/ieee802_11.c
+++ b/src/ap/ieee802_11.c
@@ -1544,6 +1544,10 @@ void ieee802_11_mgmt(struct hostapd_data *hapd, const u8 *buf, size_t len,
 		mgmt->bssid[4] == 0xff && mgmt->bssid[5] == 0xff;
 
 	if (!broadcast &&
+#ifdef CONFIG_P2P
+	    /* Invitation responses can be sent the responder MAC as BSSID */
+	    !(hapd->conf->p2p & P2P_GROUP_OWNER) &&
+#endif /* CONFIG_P2P */
 	    os_memcmp(mgmt->bssid, hapd->own_addr, ETH_ALEN) != 0) {
 		printf("MGMT: BSSID=" MACSTR " not our address\n",
 		       MAC2STR(mgmt->bssid));
-- 
1.7.0.4

